boggle_solver.js

```diff
@@ -38,70 +38,71 @@ function insert(root, word) {
     curr.is_word = true;
 }
 
-function safe(i, j, visited) {
+function safe(i, j, visited, M) {
```

Comment:
It would be better to call it isSafe to reference that the function will return a boolean